src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyIotCertificate.java

```diff
@@ -72,10 +75,11 @@ public Boolean apply(String certificatePem) {
             cert = cloudCert.get();
             if (cert.isActive()) {
                 certificateRegistry.updateCertificate(cloudCert.get());
-            } else {
-                certificateRegistry.deleteCertificate(cloudCert.get());
             }
         }
+        if (!cert.isActive()) {
+            certificateRegistry.deleteCertificate(cert);
+        }
```

Comment:
If the `cloudCert` is active at L76 but becomes inactive at L80, then we're only removing the `cloudCert` that was added in L77. The previous `cert` may still exist in store because of L75.